spec/rspec/core/hooks_filtering_spec.rb

```diff
@@ -319,6 +319,23 @@ module RSpec::Core
         expect(filters).to eq([])
       end
 
+      it "does not run :all|:context hooks if the entire context is skipped" do
+        filters = []
+        RSpec.configure do |c|
+          c.before(:all) { filters << "before all in config"}
+          c.after(:all) { filters << "after all in config"}
+          c.before(:context) { filters << "before context in config"}
+          c.after(:context) { filters << "after context in config"}
+        end
+        group = RSpec.xdescribe("skipped describe") do
+          xcontext("skipped context") do
```

Comment:
I just noticed that you are skipping both the nested group and the outer group.  Is that necessary to trigger this behavior?  It seems like it shouldn't be, but if it's not, it begs the question about why the spec is skipping at two levels.